web/services/repository_service.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Parma.RepositoryService do
+  @moduledoc """
+  Service module for performing actions on the repositories
+  """
+
+  alias Parma.Repository
+  alias Parma.User
+
+  def enable(%Repository{} = repository, %User{} = user) do
+    owner = get_repo_owner(repository)
+
+    {_status, response} = create_hook(owner, repository.name, user)
+
+    Repository.update(repository, %{enabled: true, hook_id: response|> Map.get("id") |> to_string})
+  end
+
+  def enable(_, _), do: not_found_error()
```

Comment:
is this exercised when the user or repository is nil?

Reply:
Yes, this helps avoid a `nil` check inside the function.